cardinal_toe_app/src/app/services/map-generator/map-generator.service.ts

```diff
@@ -0,0 +1,163 @@
+import { Injectable, OnInit } from '@angular/core';
+import { Location } from '../../interfaces/Location';
+import { LocationGeneratorService } from '../location-generator/location-generator.service';
+import { NumberRandomiserService } from '../number-randomiser/number-randomiser.service';
+import locationsJSON from '../../../assets/data/locations.json';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MapGeneratorService {
+  
+  // Declare constants
+  /**
+   * Sets the limit of the number of nodes (or, distance) that can be generated in the node tree
+   * away from the _'winning route'_.
+   * @private
+   * @readonly
+   * @constant
+   */
+  private readonly DEVIATION_LIMIT = 3;
+    /**
+   * Sets the limit of the number of Locations (i.e. distance) that can be generated in the Location tree
+   * before reaching the _'goal Location'_.
+   * @private
+   * @readonly
+   * @constant
+   */
+  private readonly DEPTH_LIMIT = 10;
+
+  rootLocation!: Location;
+
+  constructor ( 
+    private locationGeneratorService: LocationGeneratorService, 
+    private numberRandomiserService: NumberRandomiserService 
+  ) { }
+
+
+  getMap(): Location {
+    console.log(`Getting map...`)
+    this.generateMap();
+    return this.rootLocation;
```

Comment:
you could, to prevent accidentally regenerating the map, keep rootLocation as nullable, and check if it is null before generating the map. Like;
```
if (this.rootLocation == null) {
    this.generateMap();
}
return this.rootLocation;
```